perses/samplers/samplers.py

```diff
@@ -363,6 +371,133 @@ def has_nan(self):
             return False
 
 ################################################################################
+# GHMC INTEGRATOR (TEMPORARY UNTIL MOVED TO OPENMMTOOLS)
```

Comment:
This implementation uses an `if` block, which should work fine to protect against `nan`s. 
The [openmmtools implementation of GHMC](https://github.com/choderalab/openmmtools/blob/master/openmmtools/integrators.py#L637-L639) does not have `nan` protection yet.
Do you want me to update that first and drop the code from here in this PR, or a separate PR?